### DISCUSSION TITLE
Audio files not played

### DISCUSSION THREAD

**Discussion by Dommelen** [jspsych/jsPsych#685] [Q&A]
I wonder if it is possible to play a list of audio stimuli several times in random order. Using the code shown below doesn't work. Running the script doesn't generate any output. The crucial line seems to be "stimulus: jsPsych.timelineVariable('stimulus')," It would be great if somebody could tell me what I am doing wrong.

![Code](https://user-images.githubusercontent.com/62598051/78041912-a251f400-7371-11ea-98b6-8a70aa9efcb3.png)

**Comment by jodeleeuw** [comment]
Hi @Dommelen !

A couple things could be causing this issue.

1. You may need to preload the audio files. See this bit of the documentation on [manual preloading](https://www.jspsych.org/overview/media-preloading/#manual-preloading).

2. Some browsers now block audio from playing until the user interacts with the page. So you may need to add a simple text-based trial with a button click or keyboard response to get the user to interact before playing audio.

3. If you are running the files locally on your machine, then you can run into Cross-Origin security issues with the WebAudio player. One option here is to host the experiment on a server, like GitHub pages. Another is to use HTML5 audio instead of web audio. You can do this by setting `use_webaudio: false` in jsPsych.init():

```js
jsPsych.init({
  timeline: timeline,
  use_webaudio: false
})
```

**Comment by vijaymarupudi** [comment]
@Dommelen, could you post a .zip file with your experiment?

**Comment by Dommelen** [comment]
Thank you very much for your suggestions. Unfortunately, they don't seem to help.
Attached the code for the pilot (In the real test, I would like to present 80 stimuli 5 times each in random order).

Wim van Dommelen

[AudioPilot.zip](https://github.com/jspsych/jsPsych/files/4413470/AudioPilot.zip)

**Comment by vijaymarupudi** [comment]
[Citing the appropriate section from the documentation:](https://www.jspsych.org/overview/media-preloading/)

> If you are using media files in your experiment but they are not being passed directly as parameters to the trials (e.g., because you are using functions as parameters that return the audio, video, or images, or you are using timeline variables), then you can manually specify the files to preload.

You had to preload the audio files, as @jodeleeuw had said. I've done so in the working example attached. I've removed the use_webaudio: false line because it is not necessary in this case.

Next time, please provide the complete experiment in the zip file. It's hard to test otherwise. I've used copies of same sample wav file for demonstraction. The complete zip is attached. Here's the diff from the file you sent.

[working-example.zip](https://github.com/jspsych/jsPsych/files/4415930/working-example.zip)


```diff
22a23,29
>     var all_audio_files = [
>         "audio/word01.wav",
>         "audio/word02.wav",
>         "audio/word03.wav",
>         "audio/word04.wav"
>     ]
> 
60a68
> 
66,68c74,78
<         use_webaudio: false,
<         on_finish: function(){jsPsych.data.displayData();},
<         on_finish: function(){jsPsych.data.get().localSave('csv','Audiodata.csv');},
---
>         preload_audio: all_audio_files,
>         on_finish: function(){
>           jsPsych.data.displayData();
>           jsPsych.data.get().localSave('csv','Audiodata.csv');
>         }
71c81
< </html>
\ No newline at end of file
---
> </html>
```

**Comment by Dommelen** [comment]
Now it works! I used your code with one exception: The line "use_webaudio: false," had to be included.
Thank you very much indeed!
